worlds/alttp/UnderworldGlitchRules.py

```diff
@@ -89,13 +89,13 @@ def underworld_glitches_rules(world, player):
     # Build the rule for SP moat. 
     # We need to be able to s+q to old man, then go to either Mire or Hera at either Hera or GT. 
     # First we require a certain type of entrance shuffle, then build the rule from its pieces. 
-    if not world.worlds[player].swamp_patch_required:
-        if world.entrance_shuffle[player] in ['vanilla', 'dungeons_simple', 'dungeons_full', 'dungeons_crossed']:
+    if not world.worlds[player].options.swamp_patch_required:
```

Comment:
This needs to be reverted - swamp_patch_required is an ALTTPWorld attribute, not an option